hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieDataSourceConfig.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.config;
+
+import org.apache.hudi.common.config.DefaultHoodieConfig;
+import org.apache.hudi.common.model.OverwriteWithLatestAvroPayload;
+import org.apache.hudi.keygen.SimpleAvroKeyGenerator;
+import org.apache.hudi.keygen.constant.KeyGeneratorOptions;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Properties;
+
+public class HoodieDataSourceConfig extends DefaultHoodieConfig {
+
+  public static final String TABLE_NAME_PROP = HoodieWriteConfig.TABLE_NAME;
+  public static final String PRECOMBINE_FIELD_PROP = "hoodie.datasource.write.precombine.field";
+  public static final String RECORDKEY_FIELD_PROP = KeyGeneratorOptions.RECORDKEY_FIELD_OPT_KEY;
+  public static final String PARTITIONPATH_FIELD_PROP = KeyGeneratorOptions.PARTITIONPATH_FIELD_OPT_KEY;
+
+  public static final String WRITE_PAYLOAD_CLASS = "hoodie.datasource.write.payload.class";
+  public static final String DEFAULT_WRITE_PAYLOAD_CLASS = OverwriteWithLatestAvroPayload.class.getName();
+  public static final String KEYGENERATOR_CLASS_PROP = "hoodie.datasource.write.keygenerator.class";
+  public static final String DEFAULT_KEYGENERATOR_CLASS = SimpleAvroKeyGenerator.class.getName();
```

Comment:
Most of these options have been defined in `DataSourceOptions`, I have been planning to move them to `hudi-client-common` (https://issues.apache.org/jira/browse/HUDI-1438), till then we can reuse these options.
Do you mind waiting for a while, we can continue this PR when HUDI-1438 is finished
WDYT ？

Reply:
I agree that we should move `DataSourceOptions` to `hudi-client-common`. Are we looking for moving them into`HoodieWriteConfig`? If so I can do it on this PR. 
IMO we need to write tests for every single Flink-related feature actually, otherwise, the features are not trustworthy, so I think we should land this asap and force unit tests for the upcoming Flink features.

Reply:
> I agree that we should move `DataSourceOptions` to `hudi-client-common`. Are we looking for moving them into`HoodieWriteConfig`? If so I can do it on this PR.
> IMO we need to write tests for every single Flink-related feature actually, otherwise, the features are not trustworthy, so I think we should land this asap and force unit tests for the upcoming Flink features.

I think `DataSourceOptions` should be independent of `HoodieWriteConfig` just as it is before.
@liujinhui1994 do you have time for HUDI-1438 recently? if not, do you mind resign it to @garyli1019

Reply:
@garyli1019 @wangxianghu  

I'm almost done, I will post it as soon as possible

Reply:
@liujinhui1994 awesome!

Reply:
hi @liujinhui1994 any update on refactoring `DataSourceOptions` ticket?